packages/api/src/controllers/sensorController.js

```diff
@@ -172,13 +174,32 @@ const sensorController = {
         );
 
         // Save sensors in database
-        const sensorLocations = await Promise.all(
+        const sensorLocations = await Promise.allSettled(
           registeredSensors.map(async (sensor) => {
-            return await SensorModel.createSensor(sensor, farm_id, user_id);
+            return await SensorModel.createSensor(
+              sensor,
+              farm_id,
+              user_id,
+              esids.includes(sensor.external_id) ? 1 : 0,
+            );
           }),
         );
 
-        if (success.length + already_owned.length < esids.length) {
+        const successSensors = sensorLocations.reduce((prev, curr) => {
+          if (curr.status === 'fulfilled') {
+            prev.push(curr.value.sensor.external_id);
+          } else {
+            errorSensors.push({
+              row: data.findIndex((elem) => elem === curr.value.sensor.external_id) + 2,
```

Comment:
`curr.value.sensor` may be undefined in this case because a rejected promise will return an error instead of the sensor. maybe you can do this by index instead since `registeredSensors` and `sensorLocations` should be the same length.

Reply:
Good catch, and great idea for the solution. Thanks!